test/core/credentials/transport/alts/alts_security_connector_test.cc

```diff
@@ -117,7 +119,46 @@ TEST(AltsSecurityConnectorTest, MissingSecurityLevelPropertyFailure) {
   tsi_peer_destruct(&peer);
 }
 
-TEST(AltsSecurityConnectorTest, MissingTransportProtocol) {
+TEST(AltsSecurityConnectorTest, UnknownPeerPropertyFailure) {
+  tsi_peer peer;
+  ASSERT_EQ(tsi_construct_peer(kTsiAltsNumOfPeerProperties, &peer), TSI_OK);
+  ASSERT_EQ(tsi_construct_string_peer_property_from_cstring(
+                TSI_CERTIFICATE_TYPE_PEER_PROPERTY, TSI_ALTS_CERTIFICATE_TYPE,
+                &peer.properties[0]),
+            TSI_OK);
+  ASSERT_EQ(tsi_construct_string_peer_property_from_cstring(
+                "unknown", "alice", &peer.properties[1]),
+            TSI_OK);
+  grpc_core::RefCountedPtr<grpc_auth_context> ctx =
+      grpc_alts_auth_context_from_tsi_peer(&peer);
+  ASSERT_EQ(ctx, nullptr);
+  tsi_peer_destruct(&peer);
+}
+
+static bool test_identity(const grpc_auth_context* ctx,
```

Comment:
Can we name this something different? It looks like it's testing various peer properties, not specifically identities?